### PR TITLE
chore(maintenance): add markdownlint

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

This PR introduces markdown linting to the project. This will enforce an objective baseline for style and format of markdown files.

To support this I have added [markdownlint-cli2](https://github.com/DavidAnson/markdownlint-cli2) to the workspace, this is the same CLI used by Powertools for AWS Lambda (Python) from which I also copied the rule set.

Linting is run automatically as `pre-commit` hook via `lint-staged`, which will lint and fix only on staged files that are about to be committed (remember to run `npx husky` to make it take effect), and also as PR check. The latter runs on all markdown files in scope.

The tool supports an ignore file, which I also added to the project, however I was able to make this file work only with the IDE extension but not the CLI, for this reason I replicated the patterns in the command I added to the `package.json`.

For the time being I have ignored the `package/*/README.md` files since there are opened issues already for these, once we tackle them we will remove them from the ignore list.

Finally, if you are using VSCode, I recommend to install the [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) which will surface the issues according to the ruleset defined and can also be configured to format on save.

### Related issues, RFCs

**Issue number:** #2399

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  # Number of characters for code blocks
+  code_block_line_length: 265
+  # Include code blocks
+  code_blocks: true
+  # Include tables
+  tables: false
+  # Include headings
+  headings: true
+  # Include headings
+  headers: true
```
Comment (am29d) [inline .markdownlint.yaml:77]: My ide complains that this field is not allowed: 

Schema validation: Property 'headers' is not allowed. Looking at the comment it seems to be redundant with `headings`?

File: .markdownlint.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  # Blank lines below heading
+  lines_below: 1
+
+# MD023/heading-start-left/header-start-left - Headings must start at the beginning of the line
+MD023: true
+
+# MD024/no-duplicate-heading/no-duplicate-header - Multiple headings with the same content
+MD024:
+  # Only check sibling headings
+  allow_different_nesting: false
```
Comment (am29d) [inline .markdownlint.yaml:111]: This property seems also not valid based on the schema. I have double checked the [json schema](https://github.com/DavidAnson/markdownlint/blob/b2305efafb034b1f328845aec9928b5363ffd646/schema/markdownlint-config-schema.json#L591) and I can't find it there either.

**Review by am29d — CHANGES_REQUESTED** [review]
Great addition! I tested few formatting errors and the error messages are spot on what's wrong, where and what to fix ✅.

Just two comments on the configuration, it seems like they are not valid based on the JSON schema of markdownlint? Maybe I am missing something.

**Comment by dreamorosi** [comment]
No, you're right - I copied the config from the Python repo assuming it was correct but the two settings you pointed out are either missing or have been removed, they don't appear anymore in the [docs for the tool](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md) either.

I have removed them both!

**Review by am29d — APPROVED** [review]

**Pull request merged**